google/cloud/forseti/services/inventory/cai_temporary_storage.py

```diff
@@ -0,0 +1,421 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Inventory temporary storage for Cloud Asset data."""
+import json
+import os
+import enum
+import tempfile
+
+from retrying import retry
+from sqlalchemy import Column
+from sqlalchemy import Enum
+from sqlalchemy import Index
+from sqlalchemy import PrimaryKeyConstraint
+from sqlalchemy import String
+from sqlalchemy import LargeBinary
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.pool import SingletonThreadPool
+
+from google.cloud.forseti.common.util import logger
+from google.cloud.forseti.services.dao import create_engine
+from google.cloud.forseti.services.inventory.base.gcp import AssetMetadata
+
+LOGGER = logger.get_logger(__name__)
+BASE = declarative_base()
+MAX_ALLOWED_PACKET = 32 * 1024 * 1024  # 32 Mb default mysql max packet size
```

Comment:
This needs to be updated for sqlite, the default setting is 1mb and can be maximized to 1gb if you customize the value of the flag.
The maximum number of bytes in the text of an SQL statement is limited to SQLITE_MAX_SQL_LENGTH which defaults to 1000000. You can redefine this limit to be as large as the smaller of SQLITE_MAX_LENGTH and 1073741824.

Reply:
sqlite doesn't really use packets as such, so this is more of the limit of the number of inserts in a single execute call, I've updated the constant name and documentation.

the MAX_SQL_LENGTH applies to the statement portion, not the bound values, and as such we are far under the 1MB size.